pkg/cmd/config.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2020 The MayaData Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"github.com/spf13/pflag"
+
+	clientset "github.com/mayadata-io/dmaas-operator/pkg/generated/clientset/versioned"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Config interface provides api to access clientset and namespace
+type Config interface {
+	// BindFlags binds common flags (--kubeconfig, --namespace) to the passed-in FlagSet.
+	BindFlags(flags *pflag.FlagSet)
+	// Client returns dmaas-operator client.
+	Client() (clientset.Interface, error)
+	// KubeClient returns Kubernetes client.
+	KubeClient() (kubernetes.Interface, error)
+	// SetClientQPS sets the Queries Per Second for a client.
+	SetClientQPS(float32)
+	// SetClientBurst sets the Burst for a client.
+	SetClientBurst(int)
+	// GetNamespace return dmaas namespace
+	GetNamespace() string
+}
+
+type config struct {
+	kubeconfig  string
+	namespace   string
+	clientQPS   float32
+	clientBurst int
+	flags       *pflag.FlagSet
+}
+
+// NewConfig return config for clientset
+func NewConfig() Config {
+	c := &config{
+		flags: pflag.NewFlagSet("", pflag.ContinueOnError),
+	}
+
+	c.namespace = "dmaas"
```

Comment:
what is this namespace for? there are already few namespaces defined in other place.

Reply:
This is for default namespace. Will replace it with env variable value.